test/test_slots.py

```diff
@@ -419,6 +419,10 @@ def test_buffer(self):
                 assert sys.getrefcount(arr) == init_refcount + 1
             for i in range(12):
                 assert mv[i] == i
+        del mv
+        import gc
+        for i in range(3):
+            gc.collect()
```

Comment:
Why do we need to do `del mv`? I would have thought that the memoryview would cease to link to `arr` once the with block exits? In fact it must so that `mv2` can be created lower down, no?

It's also a bit unclear to me why the `gc.collect()` is needed here? Isn't `self.supports_refcounts()` sufficient?

Reply:
We need the gc to collect `mv` since on PyPy the buffer is only released when the gc calls the finalizer.

Reply:
But shouldn't the context manager for `mv` release the buffer when the context is left independent of the GC? Isn't that the point of having a context manager for memoryviews? Or am I not understanding something?

Reply:
Maybe this is a bug in PyPy. I will back it out, in the interest of finishing this Pr, and we can visit it again in the future.

Reply:
I found the problem in PyPy - indeed it was a bug